clang/labs/lab1/simonyan_suren/AlwaysInline.cpp

```diff
@@ -0,0 +1,85 @@
+#include "clang/AST/ASTConsumer.h"
+#include "clang/AST/Attr.h"
+#include "clang/AST/RecursiveASTVisitor.h"
+#include "clang/Frontend/CompilerInstance.h"
+#include "clang/Frontend/FrontendPluginRegistry.h"
+#include <stack>
+
+class AlwaysInlineVisitor {
+public:
+  AlwaysInlineVisitor(clang::ASTContext *MyContext) : MyContext(MyContext) {}
+
+  void VisitFunctionDecl(clang::FunctionDecl *Func) {
+    bool ContainsConditional = false;
+    std::stack<clang::Stmt *> stack;
+    stack.push(Func->getBody());
+
+    while (!stack.empty()) {
+      clang::Stmt *CurrentNode = stack.top();
+      stack.pop();
+
+      if (clang::isa<clang::IfStmt>(CurrentNode) ||
+          clang::isa<clang::SwitchStmt>(CurrentNode) ||
+          clang::isa<clang::ForStmt>(CurrentNode) ||
+          clang::isa<clang::WhileStmt>(CurrentNode) ||
+          clang::isa<clang::DoStmt>(CurrentNode)) {
+        ContainsConditional = true;
+        break;
+      }
+
+      if (auto parent = clang::dyn_cast<clang::CompoundStmt>(CurrentNode)) {
+        for (auto Child : parent->body()) {
+          stack.push(Child);
+        }
+      }
+    }
+
+    if (!ContainsConditional) {
+      clang::SourceRange FuncRange = Func->getSourceRange();
+      Func->addAttr(
+          clang::AlwaysInlineAttr::CreateImplicit(*MyContext, FuncRange));
+      Func->getAttr<clang::AlwaysInlineAttr>();
```

Comment:
For what purpose it is called?

Reply:
Forgot to remove it when checking the logs